### ISSUE TITLE
Using binary serialization where we can

### ISSUE THREAD

**Issue by skearnes** [Open-Reaction-Database/ord-schema#457]
I ran some timings with the dataset from https://github.com/Open-Reaction-Database/ord-data/pull/14, which has ~4300 reactions:
* Loading from `pbtxt`: 22s
* Loading from `pb`: 0.23s

This is especially noticeable when reviewing large datasets, since the web editor must load the entire Dataset to get to any contained reaction.

As a short-term fix I'm planning to modify the reviewer mode so that it keeps binary versions of pull request datasets locally for quick loading in the web editor. However, I'm wondering if we should also consider committing a binary version of each dataset alongside the human-readable PBTXT (which will still be considered the ground truth) in the `ord-data` repo.

WDYT?

**Comment by antonkast-google** [comment]
This strikes me as a form of caching, and I would be cautious.

Is there any value at all in using pbtxt as an internal format within the editor? Why not all binary all the time?

You may want to see my coming db/ -> Postgres refactor before optimizing the reviews. I'll have a PR on Monday.

**Comment by skearnes** [comment]
I agree we need to be cautious with caching, but if we store pb and pbtxt alongside one another this should be less of an issue and the benefits are quite significant.

There's no reason to use pbtxt internally in the editor as long as "download" always gives the user pbtxt. But I think that except for large enumerations the use of pbtxt in the editor isn't a big deal, and there's a bigger risk of cache drift if the proto changes since we don't have a canonical pbtxt alongside. Moving to binary on the backend wouldn't fix the issue we're having with reviews.

**Comment by antonkast-google** [comment]
As part of the editor-Postgres refactor, I did away with using files for datasets, including in the review mode.

Now the /review page has a "sync" button that slurps open PRs into Postgres without touching the file system. If the editor uses binaries, then so will reviews, except when you click "sync".

Would that address this problem?

**Comment by skearnes** [comment]
Yes, that would fix the problem; looking forward to your PR. In the meantime I'm going to increase the timeout on the web server to handle loading these large datasets.

**Comment by skearnes** [comment]
(As a side comment: it is _extremely_ unlikely that we will modify the protos in backwards-incompatible ways, so the risk of drift in the editor backend with binary encoding is quite low.)
